internal/allocation/allocation_manager.go

```diff
@@ -103,15 +116,24 @@ func (m *Manager) CreateAllocation(fiveTuple *FiveTuple, turnSocket net.PacketCo
 	if a := m.GetAllocation(fiveTuple); a != nil {
 		return nil, fmt.Errorf("%w: %v", errDupeFiveTuple, fiveTuple)
 	}
-	a := NewAllocation(turnSocket, fiveTuple, m.log)
+	a := NewAllocation(turnSocket, fiveTuple, m.log, requestedTransportProtocol)
 
-	conn, relayAddr, err := m.allocatePacketConn("udp4", requestedPort)
-	if err != nil {
-		return nil, err
-	}
+	if a.RequestedTransportProtocol == proto.ProtoTCP {
+		listener, relayAddr, err := m.allocateListener("tcp", requestedPort)
+		if err != nil {
+			return nil, err
+		}
+		a.RelayListener = listener
+		a.RelayAddr = relayAddr
+	} else {
+		conn, relayAddr, err := m.allocatePacketConn("udp4", requestedPort)
```

Comment:
Not a change you introduced, but I am wondering why we are listening on IPv4 only..
Seems like we are not supporting `REQUESTED-ADDRESS-FAMILY` at all..

Another topic for another PR :D